tests/unit/plugins/module_utils/test_query.py

```diff
@@ -0,0 +1,142 @@
+        result = query.parse_query([{"short_description": "LIKE SAP"}])
+
+        assert result == ([{"short_description": ("LIKE", "SAP")}], [])
+
+    def test_and_parse_query(self):
+        result = query.parse_query(
+            [{"description": "LIKE email", "short_description": "LIKE SAP"}]
+        )
+
+        assert result == (
+            [{"description": ("LIKE", "email"), "short_description": ("LIKE", "SAP")}],
+            [],
+        )
+
+    def test_or_parse_query(self):
+        result = query.parse_query(
+            [{"short_description": "LIKE SAP"}, {"description": "LIKE email"}]
+        )
+
+        assert result == (
+            [
+                {"short_description": ("LIKE", "SAP")},
+                {"description": ("LIKE", "email")},
+            ],
+            [],
+        )
+
+    def test_and_or_parse_query(self):
+        result = query.parse_query(
+            [
+                {"caller": "= abel.tuter", "state": "= new"},
+                {"caller": "= bertie.luby", "state": "= new"},
+            ]
+        )
+
+        assert result == (
+            [
+                {"caller": ("=", "abel.tuter"), "state": ("=", "new")},
+                {"caller": ("=", "bertie.luby"), "state": ("=", "new")},
+            ],
+            [],
+        )
+
+    def test_error_parse_query(self):
+        result = query.parse_query(
+            [
+                {"state": "== new"},
+            ]
+        )
+
+        assert result == ([], ["Invalid condition '== new' for column 'state'."])
+
+
+class TestSerializeQuery:
+    def test_serialize_query(self):
+        result = query.serialize_query([{"short_description": ("LIKE", "SAP")}])
+
+        assert result == "short_descriptionLIKESAP"
+
+    def test_and_serialize_query(self):
+        result = query.serialize_query(
+            [{"description": ("LIKE", "email"), "short_description": ("LIKE", "SAP")}]
+        )
+
+        assert set(result.split("^")) == set(
+            ("descriptionLIKEemail", "short_descriptionLIKESAP")
+        )
+
+    def test_or_serialize_query(self):
+        result = query.serialize_query(
+            [
+                {"short_description": ("LIKE", "SAP")},
+                {"description": ("LIKE", "email")},
+            ]
+        )
+
+        assert result == "short_descriptionLIKESAP^NQdescriptionLIKEemail"
+
+    def test_and_or_serialize_query(self):
+        result = query.serialize_query(
+            [
+                {"caller": ("=", "abel.tuter"), "state": ("=", "new")},
+                {"caller": ("=", "bertie.luby"), "state": ("=", "new")},
+            ]
+        )
+
+        assert set(result.replace("^", " ").replace("NQ", " ").split()) == set(
+            ("caller=abel.tuter", "state=new", "caller=bertie.luby")
+        )
+
+
+class TestMapQueryValues:
+    def test_map_query_values(self, mocker):
+        mapper_mock = mocker.MagicMock()
+        mapper_mock.to_snow.return_value = dict(state="1")
```

Comment:
Is there a reason why we are not using the real thing here? Mapper instances have no side effects, so I fail to see any obvious reason for mock introduction here.